packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -639,13 +646,37 @@ class _UiKitViewState extends State<UiKitView> {
       layoutDirection: _layoutDirection!,
       creationParams: widget.creationParams,
       creationParamsCodec: widget.creationParamsCodec,
+      onFocus: () {
+        _focusNode!.requestFocus();
+      }
     );
     if (!mounted) {
       controller.dispose();
       return;
     }
     widget.onPlatformViewCreated?.call(id);
-    setState(() { _controller = controller; });
+    setState(() {
+      _controller = controller;
+      _id = id;
+    });
+  }
+
+  void _onFocusChange(bool isFocused) {
+    if (!isFocused) {
+      // Unlike Android, we do not need to send "clearFocus" channel message
+      // to the engine, because focusing on another view will automatically
+      // cancel the focus on the previously focused platform view.
+    } else {
+      SystemChannels.textInput.invokeMethod<void>(
+        'TextInput.setPlatformViewClient',
+        <String, dynamic>{'platformViewId': _id},
+      ).catchError((dynamic e) {
+        // TODO(huan): remove this once "TextInput.setPlatformViewClient" is implemented on engine.
```

Comment:
Can you instead land this after the engine change?

Reply:
If we land engine first, there would be a similar problem when the engine sends "focusView" channel message that's not implemented in framework yet.

Reply:
Maybe split the PRs a little more so the landing orders would be:
1. Framework receive method channel (it will be a non-op after landing it as the method channel will never be fired from engine)
1. Engine implementation.
1. Framework send method channel 

This is a little more complicated but we can ensure there is no bad version or TODOs like this.

Reply:
@cyanglaz I think i can do that.